### PR TITLE
Nemo readme revisions

### PR DESCRIPTION
**Pull request by jgerh**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -451,12 +403,12 @@ Transformer Engine requires PyTorch to be built with at least CUDA 11.8.
 Megatron Core
 ~~~~~~~~~~~~~
 
-The NeMo LLM Multimodal Domains require that NVIDIA Megatron Core to be installed.
-Megatron core is a library for scaling large transfromer base models. 
-NeMo LLM and Multimodal models leverage Megatron Core for model parallelism, 
-transformer architectures, and optimized pytorch datasets.
+Megatron Core is required for LLM and MM domains. Although Megatron Core is pre-installed in the NVIDIA PyTorch container, you may need to update it to a newer version.
```
Comment (ericharper) [inline README.rst:406]:
```suggestion
Megatron Core is required for LLM and MM domains.
```

File: README.rst [hunk 2]
```diff
@@ -108,154 +108,111 @@ Latest News
 Introduction
 ------------
 
-NVIDIA NeMo Framework is a generative AI framework built for researchers and pytorch developers
-working on large language models (LLMs), multimodal models (MM), automatic speech recognition (ASR),
-and text-to-speech synthesis (TTS).
-The primary objective of NeMo is to provide a scalable framework for researchers and developers from industry and academia
-to more easily implement and design new generative AI models by being able to leverage existing code and pretrained models.
+NVIDIA NeMo Framework is a scalable and cloud-native generative AI framework built for researchers and PyTorch developers working on `Large Language Models <nemo/collections/nlp/README.md>`_ (LLMs), `Multimodal Models <nemo/collections/multimodal/README.md>`_ (MMs), `Automatic Speech Recognition <nemo/collections/asr/README.md>`_ (ASR), `Text to Speech <nemo/collections/tts/README.md>`_ (TTS), and `Computer Vision <nemo/collections/vision/README.md>`_ (CV). It is designed to help you efficiently create, customize, and deploy new generative AI models by leveraging existing code and pre-trained model checkpoints.
```
Comment (titu1994) [inline README.rst:111]: Please pull out the domain specific readmes at the end of the main readme. We have had questions about ASR features and models we support at ICASSP this year already due to the hiding of the ASR domain features inside if `nemo/collections/asr/README.md`.

File: README.rst [hunk 3]
```diff
@@ -108,154 +108,111 @@ Latest News
...
 
-For technical documentation, please see the `NeMo Framework User Guide <https://docs.nvidia.com/nemo-framework/user-guide/latest/playbooks/index.html>`_.
```
Comment (titu1994) [inline README.rst:117]: Please add back the link to the documentation at the very top - it is impossible to find documentation in a wall of text somewhere in the middle.

File: README.rst [hunk 4]
```diff
@@ -108,154 +108,111 @@ Latest News
...
+############################################
 
-All NeMo models are trained with `Lightning <https://github.com/Lightning-AI/lightning>`_ and
-training is automatically scalable to 1000s of GPUs.
+All NeMo models are trained with `Lightning <https://github.com/Lightning-AI/lightning>`_.
+Training is automatically scalable to 1000s of GPUs.
 
-When applicable, NeMo models take advantage of the latest possible distributed training techniques,
-including parallelism strategies such as
+When applicable, NeMo models leverage cutting-edge distributed training techniques, incorporating `parallelism strategies <https://docs.nvidia.com/nemo-framework/user-guide/latest/modeloverview.html>`_ to enable efficient training of very large models. These techniques include Tensor Parallelism (TP), Pipeline Parallelism (PP), Fully Sharded Data Parallelism (FSDP), Mixture-of-Experts (MoE), and Mixed Precision Training with BFloat16 and FP8, as well as others.
```
Comment (titu1994) [inline README.rst:119]: Be explicit - only NeMo LLM and Multimodal Models can leverage parallel strategies like above

File: README.rst [hunk 5]
```diff
@@ -108,154 +108,111 @@ Latest News
...
-and text-to-speech synthesis (TTS).
-The primary objective of NeMo is to provide a scalable framework for researchers and developers from industry and academia
-to more easily implement and design new generative AI models by being able to leverage existing code and pretrained models.
+NVIDIA NeMo Framework is a scalable and cloud-native generative AI framework built for researchers and PyTorch developers working on `Large Language Models <nemo/collections/nlp/README.md>`_ (LLMs), `Multimodal Models <nemo/collections/multimodal/README.md>`_ (MMs), `Automatic Speech Recognition <nemo/collections/asr/README.md>`_ (ASR), `Text to Speech <nemo/collections/tts/README.md>`_ (TTS), and `Computer Vision <nemo/collections/vision/README.md>`_ (CV). It is designed to help you efficiently create, customize, and deploy new generative AI models by leveraging existing code and pre-trained model checkpoints.
 
-For technical documentation, please see the `NeMo Framework User Guide <https://docs.nvidia.com/nemo-framework/user-guide/latest/playbooks/index.html>`_.
+Model Training, Alignment, and Customization
```
Comment (titu1994) [inline README.rst:113]: Be explicit - LLM Model Training, ALignment and Customization

File: README.rst [hunk 6]
```diff
@@ -108,154 +108,111 @@ Latest News
...
+NeMo LLMs can be aligned with state-of-the-art methods such as SteerLM, Direct Preference Optimization (DPO), and Reinforcement Learning from Human Feedback (RLHF). See `NVIDIA NeMo Aligner <https://github.com/NVIDIA/NeMo-Aligner>`_ for more information.
 
-NeMo's Transformer based LLM and Multimodal models leverage `NVIDIA Transformer Engine <https://github.com/NVIDIA/TransformerEngine>`_ for FP8 training on NVIDIA Hopper GPUs
-and leverages `NVIDIA Megatron Core <https://github.com/NVIDIA/Megatron-LM/tree/main/megatron/core>`_ for scaling transformer model training.
+In addition to supervised fine-tuning (SFT), NeMo also supports the latest parameter efficient fine-tuning (PEFT) techniques such as LoRA, P-Tuning, Adapters, and IA3. Refer to the `NeMo Framework User Guide <https://docs.nvidia.com/nemo-framework/user-guide/latest/sft_peft/index.html>`_ for the full list of supported models and techniques.
 
-NeMo LLMs can be aligned with state of the art methods such as SteerLM, DPO and Reinforcement Learning from Human Feedback (RLHF),
-see `NVIDIA NeMo Aligner <https://github.com/NVIDIA/NeMo-Aligner>`_ for more details.
+Model Deployment and Optimization
```
Comment (titu1994) [inline README.rst:127]: LLM Deployment and Optimization

File: README.rst [hunk 7]
```diff
@@ -108,154 +108,111 @@ Latest News
...
-------------
+Documentation
+-------------
 
-* `Large Language Models <nemo/collections/nlp/README.md>`_
-* `Multimodal <nemo/collections/multimodal/README.md>`_
-* `Automatic Speech Recognition <nemo/collections/asr/README.md>`_
-* `Text to Speech <nemo/collections/tts/README.md>`_
-* `Computer Vision <nemo/collections/vision/README.md>`_
+* `NeMo Developer Docs < https://docs.nvidia.com/nemo-framework/user-guide/latest/nemotoolkit/index.html>`_
```
Comment (titu1994) [inline README.rst:158]: Please bring up and flatten the domain readmes at the end of Documentation section (or end of readme, whatever is preferred)

Comment (titu1994) [reply README.rst:158]: Also, its unnecessary to remove the links to the doc links to the domains from a nice orderly list here and put it inline into a wall of text above in my opinion

File: README.rst [hunk 8]
```diff
@@ -108,154 +108,111 @@ Latest News
...
-2) Pytorch 1.13.1 or above
-3) NVIDIA GPU, if you intend to do model training
-
-Developer Documentation
------------------------
+* Python 3.10 or above
+* Pytorch 1.13.1 or above
+* NVIDIA GPU (if you intend to do model training)
 
-.. |main| image:: https://readthedocs.com/projects/nvidia-nemo/badge/?version=main
```
Comment (titu1994) [inline README.rst:182]: Add this back

File: README.rst [hunk 9]
```diff
@@ -108,154 +108,111 @@ Latest News
...
-| Latest  | |main|      | `Documentation of the latest (i.e. main) branch. <https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/>`_                  |
-+---------+-------------+------------------------------------------------------------------------------------------------------------------------------------------+
-| Stable  | |stable|    | `Documentation of the stable (i.e. most recent release) branch. <https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/>`_ |
-+---------+-------------+------------------------------------------------------------------------------------------------------------------------------------------+
-
-
-Getting help with NeMo
+Install NeMo Framework
 ----------------------
-FAQ can be found on NeMo's `Discussions board <https://github.com/NVIDIA/NeMo/discussions>`_. You are welcome to ask questions or start discussions there.
```
Comment (titu1994) [inline README.rst:203]: Add this back

File: README.rst [hunk 10]
```diff
@@ -108,154 +108,111 @@ Latest News
...
 
+NeMo Framework offers multiple installation options. Choose the option that is best suited to your specific domain requirements.
 
-Installation
-------------
+* The Conda/Pip installation method is recommended for ASR and TTS domains or when using NVIDIA PyTorch container as the base. See `Conda <#conda>`_ and `Pip <#pip>`_ for instructions.
 
-The NeMo Framework can be installed in a variety of ways, depending on your needs. Depending on the domain, you may find one of the following installation methods more suitable.
```
Comment (titu1994) [inline README.rst:209]: This is not true for Non LLM/Multimodal. Please do not change this section, it has been carefully updated in the past.
Or be very explicit : 
ASR, TTS, NMT can be pip installed / conda installed
LLM, MM and CV should be used in docker container.

File: README.rst [hunk 11]
```diff
@@ -108,154 +108,111 @@ Latest News
...
+* The Docker containers installation method is recommended for LLMs, MMs, and CV domains. See `Docker Containers <#docker-containers>`_ for instructions.
 
-* Conda / Pip - Refer to the `Conda <#conda>`_ and `Pip <#pip>`_ sections for installation instructions.
+The LLM and MM domains require three additional dependencies: NVIDIA Apex, NVIDIA Transformer Engine, and NVIDIA Megatron Core. See `LLM and MM Dependencies <#llm-and-mm-dependencies>`_ for information.
 
-  * This is recommended for Automatic Speech Recognition (ASR) and Text-to-Speech (TTS) domains.
-  * When using a Nvidia PyTorch container as the base, this is the recommended installation method for all domains.
```
Comment (titu1994) [inline README.rst:214]: Revert

File: README.rst [hunk 12]
```diff
@@ -108,154 +108,111 @@ Latest News
...
+NeMo Framework offers multiple installation options. Choose the option that is best suited to your specific domain requirements.
 
-Installation
-------------
+* The Conda/Pip installation method is recommended for ASR and TTS domains or when using NVIDIA PyTorch container as the base. See `Conda <#conda>`_ and `Pip <#pip>`_ for instructions.
 
-The NeMo Framework can be installed in a variety of ways, depending on your needs. Depending on the domain, you may find one of the following installation methods more suitable.
+* The Docker containers installation method is recommended for LLMs, MMs, and CV domains. See `Docker Containers <#docker-containers>`_ for instructions.
 
-* Conda / Pip - Refer to the `Conda <#conda>`_ and `Pip <#pip>`_ sections for installation instructions.
```
Comment (titu1994) [inline README.rst:211]: Revert

File: README.rst [hunk 13]
```diff
@@ -276,9 +234,10 @@ Use this installation mode if you want the version from a particular GitHub bran
     python -m pip install git+https://github.com/NVIDIA/NeMo.git@{BRANCH}#egg=nemo_toolkit[all]
 
 
-From source
-~~~~~~~~~~~
-Use this installation mode if you are contributing to NeMo.
+NeMo GitHub Repository
```
Comment (titu1994) [inline README.rst:237]: Id prefer From Source - this is developer nomenclature for building the repository from the repository

File: README.rst [hunk 14]
```diff
@@ -287,18 +246,14 @@ Use this installation mode if you are contributing to NeMo.
     cd NeMo
     ./reinstall.sh
 
-If you only want the toolkit without additional conda-based dependencies, you may replace ``reinstall.sh``
-with ``pip install -e .`` when your PWD is the root of the NeMo repository.
+If you only want the toolkit without the additional Conda-based dependencies, you can replace ``reinstall.sh`` with ``pip install -e .`` when your PWD is the root of the NeMo repository.
 
-Mac computers with Apple silicon
+Mac Computers with Apple Silicon
```
Comment (titu1994) [inline README.rst:251]: Sidenote: This is ONLY applicable to ASR domain specifically. Maybe we should mention this somewhere

File: README.rst [hunk 15]
```diff
@@ -408,35 +363,32 @@ To install Apex, run
     git checkout $apex_commit
     pip install . -v --no-build-isolation --disable-pip-version-check --no-cache-dir --config-settings "--build-option=--cpp_ext --cuda_ext --fast_layer_norm --distributed_adam --deprecated_fused_adam --group_norm"
 
+When attempting to install Apex separately from the NVIDIA PyTorch container, you might encounter an error if the CUDA version on your system is different from the one used to compile PyTorch. To bypass this error, you can comment out the relevant line in the setup file located in the Apex repository on GitHub here: https://github.com/NVIDIA/apex/blob/master/setup.py#L32.
```
Comment (titu1994) [inline README.rst:366]: Sidenote @ericharper  can we request apex folks to remove this hardcoded check ? We almost always have to uncomment it anyway, its a pain to have to clone the repo and manually edit files to get something to work

File: README.rst [hunk 16]
```diff
@@ -467,27 +419,32 @@ NeMo LLM and Multimodal may need Megatron Core to be updated to a recent version
...
+NeMo Text Processing, specifically Inverse Text Normalization, is now a separate repository. It is located here: `https://github.com/NVIDIA/NeMo-text-processing <https://github.com/NVIDIA/NeMo-text-processing>`_.
+
+Docker Containers
 ~~~~~~~~~~~~~~~~~
-We release NeMo containers alongside NeMo releases. For example, NeMo ``r1.23.0`` comes with container ``nemo:24.01.speech``, you may find more details about released containers in `releases page <https://github.com/NVIDIA/NeMo/releases>`_.
 
-To use a pre-built container, please run
+NeMo containers are launched concurrently with NeMo version updates. For example, the release of NeMo ``r1.23.0`` comes with the container ``nemo:24.01.speech``. The latest containers are:
+
+* NeMo LLM and MM container - `nvcr.io/nvidia/nemo:24.03.framework`
```
Comment (titu1994) [inline README.rst:432]: This should be updated to the unified container if its out already

File: README.rst [hunk 17]
```diff
@@ -497,25 +454,24 @@ If you choose to work with the main branch, we recommend using NVIDIA's PyTorch
...
 
-Examples
---------
+Future Work
+-----------
 
-Many examples can be found under the `"Examples" <https://github.com/NVIDIA/NeMo/tree/stable/examples>`_ folder.
+The NeMo Framework Launcher does not currently support ASR and TTS training, but it will soon.
 
+Get Help
```
Comment (titu1994) [inline README.rst:462]: Revert - it sounds very wrong to say "Get Help" - just keep it as Contributing & Discussion

File: README.rst [hunk 18]
```diff
@@ -497,25 +454,24 @@ If you choose to work with the main branch, we recommend using NVIDIA's PyTorch
...
+---------
 
-Contributing
-------------
+FAQ can be found on the NeMo `Discussions board <https://github.com/NVIDIA/NeMo/discussions>`_. You are welcome to ask questions or start discussions on the board.
 
-We welcome community contributions! Please refer to `CONTRIBUTING.md <https://github.com/NVIDIA/NeMo/blob/stable/CONTRIBUTING.md>`_ for the process.
-
-Publications
```
Comment (titu1994) [inline README.rst:511]: Add back publications, we have a new page for research publications

File: README.rst [hunk 19]
```diff
@@ -497,25 +454,24 @@ If you choose to work with the main branch, we recommend using NVIDIA's PyTorch
...
-------------
+Contribute
+----------
 
-We provide an ever-growing list of `publications <https://nvidia.github.io/NeMo/publications/>`_ that utilize the NeMo Framework.
+We welcome community contributions! Please refer to `CONTRIBUTING.md <https://github.com/NVIDIA/NeMo/blob/stable/CONTRIBUTING.md>`_ for the process.
 
-If you would like to add your own article to the list, you are welcome to do so via a pull request to this repository's ``gh-pages-src`` branch.
-Please refer to the instructions in the `README of that branch <https://github.com/NVIDIA/NeMo/tree/gh-pages-src#readme>`_.
+To contribute an article to the collection, please submit a pull request to the ``gh-pages-src`` branch of this repository. For detailed information, please consult the README located at the `gh-pages-src branch <https://github.com/NVIDIA/NeMo/tree/gh-pages-src#readme>`_.
```
Comment (titu1994) [inline README.rst:472]: TODO: @erastorgueva-nv update this part after the PR is refactored and merged

**Review by titu1994 — CHANGES_REQUESTED** [review]
Overall text changes are nice, but it leaves many ambiguities, with respect to what features are available for each domain, so please correct those.

Separately, during a recent conference, I have had comments from researchers saying they could not find the ASR models and features supported in NeMo after 1.23 - when the previous refactor pushed all the domain docs up inside of the nemo repo - and left them completely invisible to the world.

Most people will not click links nested in a wall of text to hunt down domain features and docs. So i request that the domain docs be added to the end of the NeMo main readme

File: README.rst (hunk 3 above)
Comment (jgerh) [reply README.rst:117]: OK

File: README.rst (hunk 5 above)
Comment (jgerh) [reply README.rst:113]: OK

File: README.rst (hunk 4 above)
Comment (jgerh) [reply README.rst:119]: OK

File: README.rst (hunk 6 above)
Comment (jgerh) [reply README.rst:127]: OK

File: README.rst (hunk 7 above)
Comment (jgerh) [reply README.rst:158]: Added collection ReadMe links to Documentation section

File: README.rst (hunk 8 above)
Comment (jgerh) [reply README.rst:182]: OK

File: README.rst (hunk 9 above)
Comment (jgerh) [reply README.rst:203]: The information was added to the end of the file

File: README.rst (hunk 10 above)
Comment (jgerh) [reply README.rst:209]: OK

File: README.rst (hunk 12 above)
Comment (jgerh) [reply README.rst:211]: OK

File: README.rst (hunk 11 above)
Comment (jgerh) [reply README.rst:214]: OK

File: README.rst (hunk 13 above)
Comment (jgerh) [reply README.rst:237]: "From source" is vague. How about "Build from source repository?"

File: README.rst (hunk 14 above)
Comment (jgerh) [reply README.rst:251]: Added a note

File: README.rst (hunk 17 above)
Comment (jgerh) [reply README.rst:462]: Yes, the original text was "Getting Help." How about Access the Discussions Board and Contribute to NeMo? FYI, we try and avoid the use of gerunds (ing) in headings and replace with imperative verbs for tasks.

File: README.rst (hunk 2 above)
Comment (jgerh) [reply README.rst:111]: Revert back the Key Features link. In addition, this issue will be addressed on a separate PR.

File: README.rst (hunk 7 above)
Comment (jgerh) [reply README.rst:158]: OK

File: README.rst (hunk 8 above)
Comment (jgerh) [reply README.rst:182]: Change both links to go to the nemo developer docs pending the final links which will be available next week.   Resource is ASchilling for docs versioning for 24.05 release.

File: README.rst (hunk 9 above)
Comment (jgerh) [reply README.rst:203]: OK move to bottom of file. Change to Discussions Board.

File: README.rst (hunk 10 above)
Comment (jgerh) [reply README.rst:209]: Revert the changes. Will fix this issue in a separate PR.

File: README.rst (hunk 11 above)
Comment (jgerh) [reply README.rst:214]: Revert. Will address in a separate PR

File: README.rst (hunk 13 above)
Comment (jgerh) [reply README.rst:237]: Change to Build from Source

File: README.rst (hunk 14 above)
Comment (jgerh) [reply README.rst:251]: OK

File: README.rst (hunk 16 above)
Comment (jgerh) [reply README.rst:432]: Submit a PR to update the container version. Stet as is.

File: README.rst (hunk 17 above)
Comment (jgerh) [reply README.rst:462]: OK

File: README.rst (hunk 18 above)
Comment (jgerh) [reply README.rst:511]: OK

File: README.rst (hunk 3 above)
Comment (jgerh) [reply README.rst:117]: Create a documentation heading and add user guide link here.

File: README.rst (hunk 16 above)
Comment (jgerh) [reply README.rst:468]: Updated to 24.05

File: README.rst (hunk 8 above)
Comment (jgerh) [reply README.rst]: Revert to previous content. Pending updates by Som first week of June.

File: README.rst (hunk 19 above)
Comment (jgerh) [reply README.rst:512]: After PR is merged, Som will contact Elena to update the text.

**Pull request merged**